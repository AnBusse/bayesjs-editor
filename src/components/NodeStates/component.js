import React from 'react';
import NodeState from 'components/NodeState';

const NodeStates = ({
  id,
  states,
  stateHeight,
  stateWidth,
  ...props
}) =>
  states.map((state, index) => (
    <NodeState
      key={`${id}-${state}`}
      state={state}
      index={index}
      height={stateHeight}
      width={stateWidth}
      {...props}
    />
  ));

export default NodeStates;
