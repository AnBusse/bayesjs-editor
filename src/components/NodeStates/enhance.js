import {
  pure,
  compose,
  branch,
  renderNothing,
  withProps,
} from 'recompose';
import { has, not, pipe } from 'ramda';
import { getNodeStateHeight } from 'utils/node-size';

const hasNoStates = pipe(has('states'), not);

const enhance = compose(
  pure,
  branch(hasNoStates, renderNothing),
  withProps(({ size }) => ({
    stateHeight: getNodeStateHeight(),
    stateWidth: size.width,
  })),
);

export default enhance;
