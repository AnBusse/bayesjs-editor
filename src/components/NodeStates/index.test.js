import React from 'react';
import { shallow } from 'enzyme';
import NodeStates from './component';

const defaultProps = {
  states: ['state-1', 'state-2'],
  id: 'node-id',
  stateHeight: 20,
  stateWidth: 160,
};

const shallowComponent = (props = {}) => {
  const compProps = { ...defaultProps, ...props };

  return shallow(<NodeStates {...compProps} />);
};

describe('NodeStates Component', () => {
  it('matches snapshot', () => {
    const component = shallowComponent();

    expect(component).toMatchSnapshot();
  });
});
