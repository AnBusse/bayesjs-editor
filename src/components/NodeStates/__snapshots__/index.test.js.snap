// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeStates Component matches snapshot 1`] = `
Array [
  <NodeState
    height={20}
    index={0}
    key="node-id-state-1"
    state="state-1"
    width={160}
  />,
  <NodeState
    height={20}
    index={1}
    key="node-id-state-2"
    state="state-2"
    width={160}
  />,
]
`;
